### PR TITLE
Fix case where uninstall of WC then going to WC install page would cause redirect to WP Core with access denied

### PR DESCRIPTION
**Pull request by becdetat**
#### Changes proposed in this Pull Request

* This adds a check that the store is actually installed before allowing the redirect on the store install page
* This is needed because of an edge case where on a Business, atomic site with WooCommerce installed, then WooCommerce is uninstalled, the page to install the store would immediately redirect to a core page for WC (showing an access denied message) because the site thought that WooCommerce was still installed (because that check is done using site options, which don't seem to be updated on plugin uninstall (and it would be difficult to update them since plugin uninstall happens on another site).

#### Testing instructions

* Start a business plan site with WooCommerce installed
* Uninstall WooCommerce
* Navigate to the Store page
* It should show the install page and not redirect to a core "access denied" page

### CODE REVIEW

File: client/extensions/woocommerce/app/dashboard/index.js [hunk 1]
```diff
@@ -193,8 +195,18 @@ class Dashboard extends Component {
...
-			// Redirect to Core UI setup after finish installation.
-			this.redirectToWoocommerceSetup( selectedSite );
+			// Redirect to Core UI setup after finish installation if store is
+			// installed on this site. This check is needed because of an edge
+			// case where on a site where WooCommerce was installed then
+			// removed, then shouldRedirectAfterInstall is always true because
+			// the site option still includes WooCommerce - it isn't removed on
+			// plugin removal. See
+			// /client/extensions/woocommerce/state/sites/setup-choices/selectors.js
+			// in getFinishedInstallOfRequiredPlugins().
```
Comment (mattsherman) [inline client/extensions/woocommerce/app/dashboard/index.js]: The check makes sense, but the comment doesn't! What site option still includes WooCommerce?

**Review by mattsherman — COMMENTED** [review]
I did not test this PR, but I did look at the code.

File: client/extensions/woocommerce/app/dashboard/index.js (hunk 1 above)
Comment (becdetat) [reply client/extensions/woocommerce/app/dashboard/index.js]: It made sense when I wrote it! Let me try again, I think I can simplify it.

**Comment by ilyasfoo** [comment]
@becdetat In one of my business sites, I was unable to remove WooCommerce, but I'm not sure why:

image

Fortunately, I was able to test this on another business site

**Review by ilyasfoo — APPROVED** [review]
LGTM! 🚢

**Comment by mattsherman** [comment]
> @becdetat In one of my business sites, I was unable to remove WooCommerce, but I'm not sure why:

Interesting. I know we do that with other plugins, but I've never seen it with WooCommerce. Hmm.
